### PR TITLE
ASoC: SOF: imx: Describe SAI parameters to be sent to DSP 

### PR DESCRIPTION
**Pull request by dbaluta**
This is a follow up from https://github.com/thesofproject/linux/pull/1318. Took over the PR from @groncarolo

### CODE REVIEW

**Comment by plbossart** [comment]
SOFCI TEST

**Comment by dbaluta** [comment]
@ranj063 @plbossart Thanks for review. I just remove one extre new line to make checkpatch.pl a little happier. Please review again. 

The other checkpatch.pl warnings can be ignored.

**Comment by dbaluta** [comment]
SOFCI TEST

File: sound/soc/sof/pcm.c [hunk 1]
```diff
@@ -707,6 +707,14 @@ static int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
...
+			"channels_min: %d channels_max: %d\n",
+			channels->min, channels->max);
+		break;
+	case SOF_DAI_IMX_SAI:
+		channels->min = dai->dai_config->sai.tdm_slots;
+		channels->max = dai->dai_config->sai.tdm_slots;
+
 		dev_dbg(component->dev,
 			"channels_min: %d channels_max: %d\n",
 			channels->min, channels->max);
```
Comment (paulstelian97) [inline sound/soc/sof/pcm.c:720]: The diff from git being confusing again, was going to ask why you added the print on the previous case.

File: sound/soc/sof/topology.c [hunk 2]
```diff
@@ -2704,8 +2711,66 @@ static int sof_link_sai_load(struct snd_soc_component *scomp, int index,
...
+	dev_info(scomp->dev,
+		 "tplg: config SAI%d fmt 0x%x mclk %d width %d slots %d mclk id %d\n",
+		config->dai_index, config->format,
+		config->sai.mclk_rate, config->sai.tdm_slot_width,
+		config->sai.tdm_slots, config->sai.mclk_id);
+
+	if (config->sai.tdm_slots < 1 || config->sai.tdm_slots > 8) {
+		dev_err(scomp->dev, "error: invalid channel count for SAI%d\n",
+			config->dai_index);
+		return -EINVAL;
```
Comment (paulstelian97) [inline sound/soc/sof/topology.c:2753]: Are these counts correct for SAI?

File: sound/soc/sof/pcm.c (hunk 1 above)
Comment (dbaluta) [reply sound/soc/sof/pcm.c:720]: Yes, this is because of git diff. When applied it looks correct.

File: sound/soc/sof/topology.c (hunk 2 above)
Comment (dbaluta) [reply sound/soc/sof/topology.c:2753]: TDM slots depends on git integration. AFAIK on i.MX8 we never use more than 2 slots, but SAI supports up to 8 slots.

Comment (paulstelian97) [reply sound/soc/sof/topology.c:2753]: I see, then this should be correct then (the check should be about the hardware limits, not our own usage).

**Pull request merged**